lib/stats.dx

```diff
@@ -0,0 +1,243 @@
+'# Stats
+Probability distributions and other functions useful for statistical computing
+
+
+'## Univariate probability distributions
+Simulation and evaluation of [probability distributions](https://en.wikipedia.org/wiki/Probability_distribution). 
+
+'## Dist interface
+Distributions implement the `Dist` interface.
+
+interface Dist d a
+  log_pdf : d -> a -> Float     -- either the density function or mass function
+  log_cdf : d -> a -> Float     -- the cumulative distribution function (CDF)
```

Comment:
This is a nitpick, but CDF and friends are usually only tractable or useful for one-dimensional random variables.  Perhaps `log_pdf` and `draw` could go in `Dist`, and the rest in something like `UnivariateDist`?

Reply:
Down the road I'm hoping to implement the multivariate Normal simply as a mean-shifted, linearly transformed product of standard 1D normals.

Reply:
Thanks all for the comments - I'll do what I can. In particular, I agree with this specific suggestion, but unfortunately it is beyond my knowledge of the Dex type system. I haven't been able to find much documentation on Dex type classes. I don't think that interfaces really "inherit" or "extend" in the usual way? So if I split the interface into Dist and OrderedDist I would then have to provide two different instances for each distribution? Also, I'm not sure about the syntax for dependent interfaces. I am trying to do:
```
interface Dist d a
  log_pdf : d -> a -> Float     -- either the density function or mass function
  draw : d -> Key -> a          -- function for random draws

interface [Dist d a] OrderedDist d a
  log_cdf : d -> a -> Float     -- the cumulative distribution function (CDF)
  log_sf : d -> a -> Float      -- the survivor function (complement of CDF)
  quantile : d -> Float -> a    -- the quantile function (inverse CDF)
```
but evidently this doesn't work. Hints appreciated.

Reply:
Well, that's a bug.  Filed it as #1044.  We'll get that fixed shortly, and then what you wrote should work.